### PR TITLE
Documented p2p.Server.Broadcast

### LINKED ISSUE #1126: Document p2p.Broadcast
The current documentation for broadcast is only `Broadcast a message to the world.` This is incredibly useless given that the broadcast functionality is non trivial.

https://godoc.org/github.com/prysmaticlabs/prysm/shared/p2p#Server.Broadcast

Documentation here should outline the full usage and requirements for broadcast. Furthermore, we should expand on the p2p examples to provide better insight on how things work. This lack of documentation slow progress on features like #543 by outside contributors.

Thanks @jgimeno for the feedback!

### PR DESCRIPTION
**Pull request by kushagrasharma**
Resolves #1126 

---

Added documentation and examples for p2p.Server.Broadcast.

### CODE REVIEW

File: shared/p2p/service.go [hunk 1]
```diff
@@ -231,7 +231,18 @@ func (s *Server) Send(msg proto.Message, peer Peer) {
...
+// Broadcast publishes a message to all localized peers using gossipsub. 
+// msg must be a proto.Message that can be encoded into a byte array.  
+// It publishes the first 100 chars of msg over the msg's mapped topic. 
+// To map a messageType to a topic, use RegisterTopic.
+//
+// It logs an error if msg is not a protobuf message, 
+// if msg cannot be encoded into a byte array, 
+// or if the server is unable to publish the message over gossipsub. 
+//
+//	 msg := make(chan p2p.Message, 100) // Choose a reasonable buffer size!
```
Comment (prestonvanloon) [inline shared/p2p/service.go]: The formatting here is a bit off, can you fix it?

Comment (kushagrasharma) [reply shared/p2p/service.go]: Hey - what exactly were you referring to in the formatting?

Comment (rauljordan) [reply shared/p2p/service.go]: I would only indent the code starting at line 243 with 2 spaces to mach 245 for nicer godocs - is that what you meant @prestonvanloon ?

Comment (kushagrasharma) [reply shared/p2p/service.go]: Fixed!

**Pull request merged**